apps/ios/Shared/Views/Onboarding/CreateProfile.swift

```diff
@@ -9,175 +9,244 @@
+                HStack {
+                    Text("Your profile")
+                    let name = displayName.trimmingCharacters(in: .whitespaces)
+                    let validName = mkValidName(name)
+                    if name != validName {
+                        Spacer()
+                        Image(systemName: "exclamationmark.circle")
+                            .foregroundColor(.red)
+                            .onTapGesture {
+                                alert = .invalidNameError(validName: validName)
+                            }
+                    }
+                }
+                .frame(height: 20)
+            } footer: {
+                VStack(alignment: .leading, spacing: 8) {
+                    Text("Your profile, contacts and delivered messages are stored on your device.")
+                    Text("The profile is only shared with your contacts.")
+                }
+                .frame(maxWidth: .infinity, alignment: .leading)
+            }
+        }
+        .navigationTitle("Create your profile")
+        .alert(item: $alert) { a in userProfileAlert(a, $displayName) }
+        .onAppear() {
+            DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
+                focusDisplayName = true
             }
         }
+        .keyboardPadding()
     }
+}
+
+struct CreateFirstProfile: View {
+    @EnvironmentObject var m: ChatModel
+    @Environment(\.dismiss) var dismiss
+    @State private var displayName: String = ""
+    @FocusState private var focusDisplayName
 
     var body: some View {
         VStack(alignment: .leading) {
-            Text("Create your profile")
-                .font(.largeTitle)
-                .bold()
-                .padding(.bottom, 4)
-                .frame(maxWidth: .infinity)
-            Text("Your profile, contacts and delivered messages are stored on your device.")
-                .padding(.bottom, 4)
-            Text("The profile is only shared with your contacts.")
-                .padding(.bottom)
-            ZStack(alignment: .topLeading) {
-                if !validDisplayName(displayName) {
-                    Image(systemName: "exclamationmark.circle")
-                        .foregroundColor(.red)
-                        .padding(.top, 4)
-                }
-                textField("Display name", text: $displayName)
-                    .focused($focusDisplayName)
-                    .submitLabel(.next)
-                    .onSubmit {
-                        if canCreateProfile() { focusFullName = true }
-                        else { focusDisplayName = true }
-                    }
+            Group {
+                Text("Create your profile")
+                    .font(.largeTitle)
+                    .bold()
+                Text("Your profile, contacts and delivered messages are stored on your device.")
+                    .foregroundColor(.secondary)
+                Text("The profile is only shared with your contacts.")
```

Comment:
users confuse it with contact address
```suggestion
                Text("The profile is only shared with your SimpleX Chat contacts.")
```

Reply:
same for profile update